lib/node_modules/@stdlib/array/base/count-if/lib/main.js

```diff
@@ -0,0 +1,163 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isComplexTypedArray = require( '@stdlib/array/base/assert/is-complex-typed-array' );
+var isAccessorArray = require( '@stdlib/array/base/assert/is-accessor-array' );
+var resolveGetter = require( '@stdlib/array/base/resolve-getter' );
+var reinterpret = require( '@stdlib/strided/base/reinterpret-complex' );
+
+
+// FUNCTIONS //
+
+/**
+* Counts the number of elements in an indexed array that satisfy the provided testing function.
+*
+* @private
+* @param {Collection} x - input array
+* @param {(...args: any[]) => boolean} predicate - testing array
```

Comment:
We don't use TypeScript in our JSDoc type annotations:

```suggestion
* @param {Function} predicate - testing function
```